[
{
"cve_id": "CVE-2012-3426",
"patch_date": [
"2012-06-14 15:11:17+02:00",
"2012-05-16 23:46:48+01:00",
"2012-06-05 15:27:13+01:00",
"2012-06-14 15:11:28+02:00",
"2012-06-14 22:39:08+02:00",
"2012-05-11 15:29:39+01:00"
],
"patch_files": [
"keystone/token/backends/kvs.py",
"keystone/token/core.py",
"AUTHORS",
"keystone/token/backends/sql.py",
"tests/test_keystoneclient.py",
"keystone/service.py",
"keystone/identity/core.py"
],
"earliest_patch_date": "2012-05-11 15:29:39+01:00",
"earliest_vuln_date": "2012-01-12 12:45:08-08:00",
"vuln_files": [
"keystone/identity/core.py",
"keystone/service.py",
"tests/test_keystoneclient.py"
],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:N/AC:M/Au:S/C:P/I:P/A:N",
"accessVector": "NETWORK",
"accessComplexity": "MEDIUM",
"authentication": "SINGLE",
"confidentialityImpact": "PARTIAL",
"integrityImpact": "PARTIAL",
"availabilityImpact": "NONE",
"baseScore": 4.9
},
"severity": "MEDIUM",
"exploitabilityScore": 6.8,
"impactScore": 4.9,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-264"
],
"fix_duplicate_type": false,
"fix_duplicate_types": [],
"intro_duplicate_type": false,
"intro_duplicated_types": [],
"file_fixed_before_fixed_again": false,
"files_fixed_before_fixed_again": [],
"file_fixed_before_vuln_again": false,
"files_fixed_before_vuln_again": []
},
{
"cve_id": "CVE-2012-4456",
"patch_date": [
"2012-06-03 12:24:10-05:00",
"2012-06-26 20:18:21+02:00",
"2012-06-03 12:08:44-05:00",
"2012-06-26 20:18:09+02:00"
],
"patch_files": [
"tests/test_content_types.py",
"keystone/catalog/core.py",
"keystone/identity/core.py"
],
"earliest_patch_date": "2012-06-03 12:08:44-05:00",
"earliest_vuln_date": "None",
"vuln_files": [],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
"accessVector": "NETWORK",
"accessComplexity": "LOW",
"authentication": "NONE",
"confidentialityImpact": "PARTIAL",
"integrityImpact": "PARTIAL",
"availabilityImpact": "PARTIAL",
"baseScore": 7.5
},
"severity": "HIGH",
"exploitabilityScore": 10,
"impactScore": 6.4,
"acInsufInfo": false,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-287"
],
"fix_duplicate_type": false,
"fix_duplicate_types": [],
"intro_duplicate_type": false,
"intro_duplicated_types": [],
"file_fixed_before_fixed_again": true,
"files_fixed_before_fixed_again": [
"keystone/identity/core.py"
],
"file_fixed_before_vuln_again": false,
"files_fixed_before_vuln_again": []
},
{
"cve_id": "CVE-2012-4457",
"patch_date": [
"2012-07-30 17:43:23+02:00",
"2012-07-16 16:08:34-05:00"
],
"patch_files": [
"tests/test_keystoneclient.py",
"keystone/service.py"
],
"earliest_patch_date": "2012-07-16 16:08:34-05:00",
"earliest_vuln_date": "2012-06-20 07:39:42+08:00",
"vuln_files": [
"keystone/service.py"
],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:N/AC:L/Au:S/C:P/I:N/A:N",
"accessVector": "NETWORK",
"accessComplexity": "LOW",
"authentication": "SINGLE",
"confidentialityImpact": "PARTIAL",
"integrityImpact": "NONE",
"availabilityImpact": "NONE",
"baseScore": 4
},
"severity": "MEDIUM",
"exploitabilityScore": 8,
"impactScore": 2.9,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-287"
],
"fix_duplicate_type": true,
"fix_duplicate_types": [
"CWE-287"
],
"intro_duplicate_type": true,
"intro_duplicated_types": [
"CWE-287"
],
"file_fixed_before_fixed_again": true,
"files_fixed_before_fixed_again": [
"tests/test_keystoneclient.py",
"keystone/service.py"
],
"file_fixed_before_vuln_again": true,
"files_fixed_before_vuln_again": [
"tests/test_keystoneclient.py",
"keystone/service.py"
]
},
{
"cve_id": "CVE-2012-3542",
"patch_date": [
"2012-08-30 10:22:47-04:00",
"2012-08-23 08:02:55-05:00"
],
"patch_files": [
"keystone/identity/core.py"
],
"earliest_patch_date": "2012-08-23 08:02:55-05:00",
"earliest_vuln_date": "None",
"vuln_files": [],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:N/AC:M/Au:N/C:N/I:P/A:N",
"accessVector": "NETWORK",
"accessComplexity": "MEDIUM",
"authentication": "NONE",
"confidentialityImpact": "NONE",
"integrityImpact": "PARTIAL",
"availabilityImpact": "NONE",
"baseScore": 4.3
},
"severity": "MEDIUM",
"exploitabilityScore": 8.6,
"impactScore": 2.9,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-264"
],
"fix_duplicate_type": true,
"fix_duplicate_types": [
"CWE-264"
],
"intro_duplicate_type": false,
"intro_duplicated_types": [],
"file_fixed_before_fixed_again": true,
"files_fixed_before_fixed_again": [
"keystone/identity/core.py"
],
"file_fixed_before_vuln_again": false,
"files_fixed_before_vuln_again": []
},
{
"cve_id": "CVE-2012-5571",
"patch_date": [
"2012-11-16 14:11:10+01:00",
"2012-11-22 14:16:47+00:00",
"2012-11-13 15:49:19-08:00"
],
"patch_files": [
"keystone/contrib/ec2/core.py"
],
"earliest_patch_date": "2012-11-13 15:49:19-08:00",
"earliest_vuln_date": "2012-01-18 20:57:44-08:00",
"vuln_files": [
"keystone/contrib/ec2/core.py"
],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:N/AC:M/Au:S/C:N/I:P/A:N",
"accessVector": "NETWORK",
"accessComplexity": "MEDIUM",
"authentication": "SINGLE",
"confidentialityImpact": "NONE",
"integrityImpact": "PARTIAL",
"availabilityImpact": "NONE",
"baseScore": 3.5
},
"severity": "LOW",
"exploitabilityScore": 6.8,
"impactScore": 2.9,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-255"
],
"fix_duplicate_type": false,
"fix_duplicate_types": [],
"intro_duplicate_type": false,
"intro_duplicated_types": [],
"file_fixed_before_fixed_again": false,
"files_fixed_before_fixed_again": [],
"file_fixed_before_vuln_again": false,
"files_fixed_before_vuln_again": []
},
{
"cve_id": "CVE-2012-5563",
"patch_date": [
"2012-11-28 10:28:07-05:00",
"2012-11-28 16:44:42+01:00"
],
"patch_files": [
"keystone/service.py",
"tests/test_service.py"
],
"earliest_patch_date": "2012-11-28 10:28:07-05:00",
"earliest_vuln_date": "None",
"vuln_files": [],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:N/AC:L/Au:S/C:N/I:P/A:N",
"accessVector": "NETWORK",
"accessComplexity": "LOW",
"authentication": "SINGLE",
"confidentialityImpact": "NONE",
"integrityImpact": "PARTIAL",
"availabilityImpact": "NONE",
"baseScore": 4
},
"severity": "MEDIUM",
"exploitabilityScore": 8,
"impactScore": 2.9,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-255"
],
"fix_duplicate_type": true,
"fix_duplicate_types": [
"CWE-255"
],
"intro_duplicate_type": false,
"intro_duplicated_types": [],
"file_fixed_before_fixed_again": true,
"files_fixed_before_fixed_again": [
"keystone/service.py"
],
"file_fixed_before_vuln_again": false,
"files_fixed_before_vuln_again": []
},
{
"cve_id": "CVE-2013-0270",
"patch_date": [
"2013-01-21 19:54:29-05:00",
"2013-02-05 16:17:24+00:00"
],
"patch_files": [
"keystone/common/utils.py",
"keystone/config.py",
"tests/test_sizelimit.py",
"tests/test_service.py",
"keystone/exception.py",
"keystone/service.py",
"keystone/middleware/core.py",
"etc/keystone.conf.sample"
],
"earliest_patch_date": "2013-01-21 19:54:29-05:00",
"earliest_vuln_date": "2012-06-19 14:16:29-05:00",
"vuln_files": [
"etc/keystone.conf.sample"
],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:N/AC:L/Au:N/C:N/I:N/A:P",
"accessVector": "NETWORK",
"accessComplexity": "LOW",
"authentication": "NONE",
"confidentialityImpact": "NONE",
"integrityImpact": "NONE",
"availabilityImpact": "PARTIAL",
"baseScore": 5
},
"severity": "MEDIUM",
"exploitabilityScore": 10,
"impactScore": 2.9,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-119"
],
"fix_duplicate_type": false,
"fix_duplicate_types": [],
"intro_duplicate_type": false,
"intro_duplicated_types": [],
"file_fixed_before_fixed_again": true,
"files_fixed_before_fixed_again": [
"tests/test_service.py",
"keystone/service.py"
],
"file_fixed_before_vuln_again": true,
"files_fixed_before_vuln_again": [
"keystone/service.py"
]
},
{
"cve_id": "CVE-2013-2006",
"patch_date": [
"2013-05-01 15:14:24-07:00"
],
"patch_files": [
"keystone/common/config.py"
],
"earliest_patch_date": "2013-05-01 15:14:24-07:00",
"earliest_vuln_date": "2013-03-20 10:37:48-05:00",
"vuln_files": [
"keystone/common/config.py"
],
"impact": {
"baseMetricV2": {
"cvssV2": {
"version": "2.0",
"vectorString": "AV:L/AC:L/Au:N/C:P/I:N/A:N",
"accessVector": "LOCAL",
"accessComplexity": "LOW",
"authentication": "NONE",
"confidentialityImpact": "PARTIAL",
"integrityImpact": "NONE",
"availabilityImpact": "NONE",
"baseScore": 2.1
},
"severity": "LOW",
"exploitabilityScore": 3.9,
"impactScore": 2.9,
"obtainAllPrivilege": false,
"obtainUserPrivilege": false,
"obtainOtherPrivilege": false,
"userInteractionRequired": false
}
},
"cwes": [
"CWE-200"
],
"fix_duplicate_type": false,
"fix_duplicate_types": [],
"intro_duplicate_type": false,
"intro_duplicated_types": [],
"file_fixed_before_fixed_again": false,
"files_fixed_before_fixed_again": [],
"file_fixed_before_vuln_again": false,
"files_fixed_before_vuln_again": []
}
]